src/lib/util/coretmpl.h

```diff
@@ -351,4 +351,74 @@ class fixed_allocator
 };
 
 
+// ======================> contiguous_sequence_wrapper
+
+namespace util {
+
+// wraps an existing sequence of values
+template<typename T >
```

Comment:
The spacing is screwed up - how about using `template <typename T>` to be more like other code?